app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -842,7 +842,14 @@ private void moveOverlayUpOrDownToMatchAppraisalBox() {
             return;
         }
 
-        moveOverlay(appraisalBox.getVisibility() == View.VISIBLE);
+        if (inputBox.getVisibility() == View.VISIBLE) {
+            boolean appraisalExpanded = appraisalBox.getVisibility() == View.VISIBLE;
+            moveOverlay(appraisalExpanded);
+        } else { //always move overlay down if not on input screen. Appraisal box cant be open if user is not in
+            // input screen
+            moveOverlay(false);
+        }
+
```

Comment:
> moveOverlay(inputBox.getVisibility() == View.VISIBLE && appraisalBox.getVisibility() == View.VISIBLE);

That would check two booleans:

1. the user is viewing the input box
2. the user has expanded the appraisal box.

Checking both of those is not necessary, as it's not possible for the user to be viewing the appraisal box if the input box is not visible. This is because the appraisal box is **within** the input box, so if the input box is invisible, the appraisal box is always invisible.